android/app/src/main/java/com/foss/foss/feature/statsearching/relative/RelativeStatsViewModel.kt

```diff
@@ -38,20 +39,24 @@ class RelativeStatsViewModel(
     val event: SharedFlow<RelativeStatsEvent>
         get() = _event.asSharedFlow()
 
-    fun fetchRelativeStats() {
-        viewModelScope.launch {
-            relativeStatsRepository.fetchRelativeStats(name.value)
-                .onSuccess { relativeStats ->
-                    _relativeStats.value = relativeStats.map { it.toUiModel() }
-                }.onFailure {
-                    _event.emit(RelativeStatsEvent.Failed)
-                }
-        }
+    fun fetchRelativeStats(nickname: String) {
+        runCatching {
+            _nickname = Nickname(nickname)
+            viewModelScope.launch {
+                relativeStatsRepository.fetchRelativeStats(_nickname)
+                    .onSuccess { relativeStats ->
+                        _relativeStats.value = relativeStats.map { it.toUiModel() }
+                    }.onFailure {
+                        _event.emit(RelativeStatsEvent.Failed)
+                    }
+            }
+            // TODO: UiState로 적용하면서 닉네임 변환 실패 시 처리 필요
+        }.onFailure { }
```

Comment:
[P3]
```suggestion
    fun fetchRelativeStats(nickname: String) {
        runCatching {
            viewModelScope.launch {
                relativeStatsRepository.fetchRelativeStats(Nickname(nickname))
                    .onSuccess { relativeStats ->
                        _relativeStats.value = relativeStats.map { it.toUiModel() }
                    }.onFailure {
                        _event.emit(RelativeStatsEvent.Failed)
                    }
            }
            // TODO: UiState로 적용하면서 닉네임 변환 실패 시 처리 필요
        }.onFailure { }
```
lateinit var에 대한 변수를 두지 않고 다음과 같은 형식으로 repository 함수를 호출하는 것은 어떨까요?

Reply:
진욱님께서 제시해주신 코드 보면 ViewModelScope 으로 runCatching을 감싸고 있는데, 이러면 fetchRelativeStats()에서 실패를 한 것인지, Nickname 모델로 변경하는데 실패한 것인지 알 수 있나요 ??

Reply:
윤혁님이 말해주신 것처럼 runCatching을 외부로 뺀 이유는 `Nickname` 모델로 변경하는데 실패한 것을 나타내기 위함이었습니다.  
어떻게 구현할지는 모르겠지만 제 생각에는 `Nickname`모델로 변경할 때의 실패와 리포지터리에서 받아올 때 실패를 구분하여 처리할 것 같다는 생각을 했어요. 그래서 runCatching이 가장 밖에 있도록 구현한 것입니다!  

혹시 더 좋은 방법이 있을까요?